### ISSUE TITLE
soil moisture variable name inconsistencies

### ISSUE THREAD

**Issue by taylor13** [cmip6dr/CMIP6_DataRequest_VariableDefinitions#413]
The CORDEX project noted:

In the CORDEX-CMIP6 Atmos Variable List there are 3 variables with Soil Moisture Content (water in all phases):

- Total Soil Moisture Content (`mrso`) – “The mass of water in all phases per unit area, summed over all soil layers.”
- Moisture in Upper Portion of Soil Column (`mrsos`)  - “The mass of water in all phases in the upper 10cm of the soil layer.”
- Total Water Content of Soil Layer (`mrsol`) – “The mass of water in all phases in each soil layer (3D variable).”

These 3 variables are taken from the CMIP6 Data Request, exactly as they are (https://clipc-services.ceda.ac.uk//dreq/mipVars.html). If we follow the same logic, a similar triplet for Frozen Water Content should be `mrfso`, `mrfsos` and `mrfsol` and for Liquid Water Content - `mrlso`, `mrlsos` and `mrlsol`. There are `mrfso` and `mrlso` in the CMIP6 DR but instead of `mrfsol`/`mrlsol` one can find only `mrsfl`/`mrsll`. Both variables (`mrsfl` and `mrsll`) are only defined in the Palaeoclimate MIP and I don’t know how names for them were created. It should be checked with CMIP.

I replied:

The rules for naming MIP variables originated at the time of AMIP about 30 years ago (formulated primarily by Peter Gleckler and me).  You can find the guidance at https://pcmdi.llnl.gov/mips/amip/DATASTDS/VARNAMES/main.html .  According to those rules, we should have:
```
                           Liquid+Frozen         Liquid               Frozen
                          ----------             ------               -------
column total       |      mrso                    mrsol                mrsof
top 10 cm          |      mrsos                   mrsols               mrsofs
by layer           |                                       
```
In the early 90's we didn't anticipate needing to record soil moisture by layer. 

In CMIP3 we had 
```
                           Liquid+Frozen         Liquid               Frozen
                          ----------             ------               -------
column total       |      mrso                                         mrfso
top 10 cm          |      mrsos                  
by layer           |                         
```
We didn't exactly follow the AMIP rule for frozen water (``mrfso``) because we put the "f" before "so", rather than after it, which we should have done to be consistent with ``mrsol`` in the first table above.  

In CMIP5 we had 
```
                           Liquid+Frozen         Liquid               Frozen
                          ----------             ------               -------
column total       |      mrso                                         mrfso
top 10 cm          |      mrsos                  
by layer           |      mrlsl                   
```
We appear to have mistakenly used the name "mrlsl" to represent the total moisture content in each layer.  It should have probably been named "mrsol": moisture (content)-(of) soil-(by) layer.  This got corrected in CMIP6.

In CMIP6 we have:
```
                           Liquid+Frozen         Liquid               Frozen
                          ----------             ------               -------
column total       |      mrso                    mrlso                mrfso
top 10 cm          |      mrsos                  
by layer           |      mrsol                   mrsll                mrsfl   
```
In CMIP6 we also have ``mrsosLut`` and ``mrsoLut`` which have the additional dimension of "land use type" but are otherwise defined like ``mrsos`` and ``mrso``.  
Note that ``mrsll`` and ``mrsfl`` should probably have been ``mrlsol`` and ``mrfsol`` (without dropping the "o").   for consistency with ``mrlso`` and ``mrfso`` and the other soil moisture variables. 

I would recommend that CORDEX adopt the names from CMIP6 despite any flaws.

I would further note that we are reconsidering the rules for the naming of variables in CMIP7.  Most names will be unchanged, but we will use additional descriptors to modify the root names in a standardized way.  Distinguishing between variables representing the "top 10 cm" or "by layer" from the "column totals" would be left up to these additional descriptors and all the 3 roots (``mrso``, ``mrlso``, and ``mrfso``) would apply to to all rows in the tables above.  

Finally, if it were up to me, I would never ask anyone to record "liquid+frozen", "liquid", *and* "frozen" soil moisture.  Two of the three should be sufficient because given two, the third can be computed from subtraction or addition. 

For Frozen Water, I guess either ``mrsofs`` (consistent with the original rules and not inconsistent with ``mrsos``) or ``mrfsos`` (building on ``mrfso`` and not inconsistent with ``mrsos``) would be acceptable, but my slight preference would be ``mrfsos``, since the original rule wasn't followed in creating ``mrfso``.

**Comment by jesusff** [comment]
May I bring your attention also to soil temperature?

It is [currently coded](https://clipc-services.ceda.ac.uk//dreq/u/3fd3cb4e2d6dd15b7dd73def96fea129.html) as `tsl`. This is a 3D var stored by layer (thus, ends in `l`). In the context of the urban CORDEX FPS, we are planning to store the temperature of the top layer. Shall this be called then `tss`? What would you recommend?

Luckily, column totals (or averages) for soil temperature are not very useful, otherwise, following the rules above we would have `ts`, and this would collide with the skin temperature. All this arises because of not using consistently `so` to refer to the soil. In the `tsl` name, is seems that `sl` was meant for soil, rather than being `s`oil `l`ayers. A consistent use of `so` for the soil would lead to:
```                     
column total  | tso  (not useful)
top 10 cm     | tsos             
by layer      | tsol      
```

**Comment by taylor13** [comment]
Just to note that in CMIP6, "tsl" can either be a function of model layer or it can represent the mean over the uppermost 10 cm of soil.  The variable name normally does not need to indicate how the data have been sampled (what dimensions are covered) either spatially or temporally.   So, it would be o.k. to use "tsl" to report the temperature of a model's top soil layer.  The variable should be a function of lon, lat, and the appropriate vertical coordinate defined in the model (presumably z ... the depth below the surface).  The z-coordinate would be a scalar with a value usually set to the middle depth of the uppermost layer and with bounds set to 0 (top of soil layer) and the layer's lower bound (which presumably will vary from one model to another).  If soil layers vary with location, I'm not sure what one would do.

Would it be easier to compare models if everyone reported the temperature of the top 10 cm of soil, as in CMIP6?

**Comment by jesusff** [comment]
Yes, sorry, I meant we would save the temperature of the top 10 cm, which might not be a soil layer for all models. Saving just the top model layer is not very useful, since this top layer depth in general depends across models. One would need the rest of the layers to compute some comparable diagnostics. Our idea was to compute this top 10cm temperature diagnostic in advance and save it as an output field.

**Comment by taylor13** [comment]
I would name the temperature of the top 10 cm as in CMIP6: "tsl".  I think you're right that "sl" is an abbreviation for SoiL.  If soil temperature as a function of depth is also collected it could also be called "tsl", but it would have to appear in a different CMOR table.
